### PR TITLE
Random bogus error in relocate/failover dialog: "No DRPolicy found"

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2235423

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
image

File: packages/mco/components/modals/app-failover-relocate/subscriptions/failover-relocate-modal-body.tsx [hunk 1]
```diff
@@ -128,52 +128,30 @@ export const FailoverRelocateModalBody: React.FC<FailoverRelocateModalBodyProps>
...
-          : {},
-      [
-        subscriptions,
-        subscriptionsLoaded,
-        subscriptionsLoadError,
-        placementMap,
-        application,
-      ]
-    );
-
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/subscriptions/failover-relocate-modal-body.tsx:168]: useEffect displays NoPolicy found error message before subscriptionMap and placementMap are executed. 
Moved all parsing under useEffect and executed in order. It is App namespace-specific parsing, So the number of resources will be minimum only.

File: packages/mco/components/modals/app-data-policies-status/subscriptions/data-policies-status-popover.tsx [hunk 2]
```diff
@@ -181,7 +181,7 @@ export const DataPoliciesStatusPopover: React.FC<DataPoliciesStatusPopoverProps>
     const launchModal = React.useCallback(
       () =>
         launcher(DataPoliciesStatusModal, {
-          ...dataPoliciesStatus,
+          extraProps: dataPoliciesStatus,
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-data-policies-status/subscriptions/data-policies-status-popover.tsx:184]: Invalid argument object spread

**Comment by SanjalKatiyar** [comment]
/override ci/prow/odf-console-e2e-aws
our CI only have checks for ODF and this PR introduces changes in MCO UI, therefore overridding the CI to merge this PR (CI is blocked right now due to issue under investigation)

**Pull request merged**